data-model/src/main/java/io/micronaut/data/model/runtime/RuntimePersistentEntity.java

```diff
@@ -415,9 +415,19 @@ public RuntimePersistentProperty<T>[] getConstructorArguments() {
      */
     public boolean hasAutoPopulatedProperties() {
         if (this.hasAutoPopulatedProperties == null) {
-            this.hasAutoPopulatedProperties = Arrays.stream(allPersistentProperties)
+            boolean direct = Arrays.stream(allPersistentProperties)
                     .filter(Objects::nonNull)
                     .anyMatch(PersistentProperty::isAutoPopulated);
+            if (!direct) {
+                // Also consider embedded associated entities
+                for (RuntimeAssociation<?> association : getAssociations()) {
```

Comment:
This change is causing issues with 
```
public <T> RuntimePersistentEntity<T> getEntity(@NonNull Class<T> type) {
        ArgumentUtils.requireNonNull("type", type);
        // we need atomicity here, since entites are compared by identity (==)
        return entities.computeIfAbsent(type, this::newEntity);
    }
```
concurrent hash map updates and throws intermittently 'Recursion update' error because during entity creation it checks for `hasAutoPopulatedProperties` and triggers creation of associated entity which sometimes throws error.